src/tools/rbd/action/Import.cc

```diff
@@ -132,6 +138,30 @@ static int do_import(librbd::RBD &rbd, librados::IoCtx& io_ctx,
 #endif
   }
 
+  if (import_snaps || !from_stdin) {
+    r = safe_read_exact(fd, image_buf, utils::RBD_IMAGE_BANNER.size());
+    if (r < 0)
+      goto done;
+    image_buf[utils::RBD_IMAGE_BANNER.size()] = '\0';
+    if (strcmp(image_buf, utils::RBD_IMAGE_BANNER.c_str())) {
+      // Old format
+      if (from_stdin) {
+	// --with-snaps specified for an old format exported file
+	r = -EINVAL;
+	goto done;
+      }
+      // seek back to the beginning.
+      lseek(fd, 0, SEEK_SET);
```

Comment:
Sorry, I am not sure what's your suggestion here, but I can explain the logic here more clear.

(1).import_snaps == true means --with-snaps is specified.
(2).(from_stdin == false )means importing something from file.

Then there are 3 possibility we will into this if statement.
(1). (import_snaps == true) and (from_stdin == true), in this case, when we found it's old format, because it's from stdin, we can't seek back, so return -EINVAL. That means, you have to make sure the you are accepting a new format importing stdin when you specify --with-snaps manually. 

(2). (import_snaps == true) and (from_stdin == false), in this case, when we found it's a old format. we will lseek to the beginning, because it's from a fd. and the all data in file should be imported into image.

(3). (import_snaps == false) and (from_stdin == false), in this case, --with-snaps are not specified, and we are importing data from a file. then we try to check the banner at first, if we found it's old format. lseek to 0 to import it in old manner. 

In this way, we can keep the backforwad compatability for the old format of export/exportdiff